src/components/MapController/MapController.js

```diff
@@ -89,93 +157,220 @@ var MapController = React.createClass({
+        }
+      }
+
+      //select a starting timestamp to display.
+      //should correspond to a 0th index, but might be annual, seasonal, or monthly
+      var startingTime = Object.keys(times)[0];
+      
+      //store values and render
       this.setState({
-        times: response.data[this.selectedDataset.unique_id].times,
-        timeidx: 0,
-        dataset: this.selectedDataset.unique_id,
-        wmstime: response.data[this.selectedDataset.unique_id].times[0],
-        variable: this.selectedDataset.variable_id,
+        run: run,
+        start_date: start_date,
+        end_date: end_date,
+        times: times,
+        timeidx: startingTime, 
+        wmstime: times[startingTime],
+        variable: nextProps.meta[0].variable_id,
+        comparand: nextProps.comparandMeta ? nextProps.comparandMeta[0].variable_id: undefined,
+        scalarPalette: sPalette,
+        contourPalette: cPalette
       });
-    });
+    });    
   },
 
   shouldComponentUpdate: function (nextProps, nextState) {
     // This guards against re-rendering before we have required data
-    return JSON.stringify(nextState) !== JSON.stringify(this.state);
+    return !_.isEqual(nextState, this.state);
   },
 
-  render: function () {
-    var pallettes = [['boxfill/ferret', 'ferret'],
-                     ['boxfill/rainbow', 'rainbow'],
-                     ['boxfill/occam', 'occam'],
-                     ['boxfill/occam_inv', 'inverted occam'],
-                    ];
+  //renders a CanadaMap, menu buttons, and a dialog box with a lot of view options
+  render: function () {          
+    //populate UI selectors: palette and scale for both isolines and blocks,
+    //run and period dropdown, time of year selector, number of isolines
+    
+    //colour selector
+    var palettes = [
+      ['seq-Blues', 'light blues'],
+      ['seq-BkBu', 'dark blues'],
+      ['seq-Greens', 'light greens'],
+      ['seq-BkGn', 'dark greens'],
+      ['seq-Oranges', 'oranges'],
+      ['seq-BuPu', 'purples'],
+      ['seq-Greys', 'greys'],
+      ['seq-BkYl', 'yellows'],
+      ['x-Occam', 'rainbow'],
+      ['default', 'ocean'],
+      ['seq-cubeYF', 'cube'],
+      ['psu-magma', 'sunset']
+    ];
+    
+    //logscale selectors
     var colorScales = [['false', 'Linear'], ['true', 'Logarithmic']];
 
-    // Determine available datasets and display selector if multiple
-    //FIXME: Time period should be determined from the metadata API
-    // which currently doesn't give time bounds information. See here:
-    // https://github.com/pacificclimate/climate-explorer-backend/issues/44
-    // When that issue is fixed, this code needs to be updated
+    //run and period selector
+    //displays a list of all the unique combinations of run + climatological period
+    //a user could decide to view.
+    //Not sure JSON is the right way to do this, though.
+    //TODO: see if there's a more elegant way to handle the callback 
+    //(selector won't pass an object)
```

Comment:
Agree could be more elegant.

The obvious alternative (but not thought through here in detail) would be to use an index into a list (whether implicit -- you'd need to establish a canonical ordering for that --, or explicit -- you'd need to share a data structure for that).